macros/sql/dedupe.sql

```diff
@@ -0,0 +1,41 @@
+{%- macro dedupe(table, group_by, order_by=none) -%}
+    {{ return(adapter.dispatch('dedupe', 'dbt_utils')(table, group_by, order_by=order_by)) }}
+{% endmacro %}
+
+{%- macro default__dedupe(table, group_by, order_by=none) -%}
+
+    select
+        {{ dbt_utils.star(table, relation_alias='deduped') | indent }}
+    from (
+        select
+            _inner.*,
+            row_number() over (
+                partition by {{ group_by }}
+                {% if order_by != none -%}
+                order by {{ order_by }}
+                {%- endif %}
+            ) as rn
+        from {{ table }} as _inner
+    ) as deduped
+    where deduped.rn = 1
+
+{%- endmacro -%}
+
+{%- macro bigquery__dedupe(table, group_by, order_by=none) -%}
+
+    select
+        {{ dbt_utils.star(table, relation_alias='deduped') | indent }}
+    from (
+        select
+            array_agg (
+                original
+                {% if order_by != none -%}
+                order by {{ order_by }}
+                {%- endif %}
+                limit 1
+            )[offset(0)] as deduped
+        from {{ table }} as original
+        group by {{ group_by }}
+    )
+
+{%- endmacro -%}
```

Comment:
It'd be useful to have a comment explaining why there's two variations here - I forgot that it provided performance benefits until I went and looked at the original issue. Even just a link to [that comment](https://github.com/dbt-labs/dbt-utils/issues/335#issuecomment-788157572) would be useful.